### DISCUSSION TITLE
Sensore allarme tipo "tenda" per Home Assistant

### DISCUSSION THREAD

**Discussion by f18m** [indomus/forum#2870] [GENERALE - Discussioni generiche]
Buongiorno,
Ho letto la guida inDomus per realizzare un impianto allarme "fai-da-te" tramite HA (https://indomus.it/progetti/realizzare-un-proprio-sistema-dallarme-antifurto-domotico-con-home-assistant/). Anzitutto grazie per le ottime guide allo staff di inDomus!!
Nella guida (e altrove) si fa riferimento principalmente a sensori di tipo PIR pensati per rilevare movimento in una stanza, oppure a sensori classici a "2 parti" per rilevare l'apertura/chiusura di porte/finestre.
Fin qui tutto OK.
Tuttavia io sarei interessato a sensori ad effetto "tenda" (es: **LINK RIMOSSO**  oppure **LINK RIMOSSO** oppure **LINK RIMOSSO** ).
Questi sensori sono utili in quanto triggerano l'allarme PRIMA dell'eventuale irruzione del ladro.

Ho cercato un pò ma tuttora non ne ho trovato NESSUNO con standard "aperto" e integrabile in HomeAssistant.
Vi chiedo aiuto: siete a conoscenza di un qualche sensore effetto "tenda" o "barriera" installabile in esterno, wireless e integrabile in HomeAssistant?

Grazie mille in anticipo!

**Comment by snakuzzo** [comment]
Rimuovo tutti i link. Sono vietati link a store. Mi spiace

**Reply by f18m** [reply]
Ops, mi era sfuggito, chiedo venia...

**Comment by KC-inDomus** [comment]
> Buongiorno, Ho letto la guida inDomus per realizzare un impianto allarme "fai-da-te" tramite HA (https://indomus.it/progetti/realizzare-un-proprio-sistema-dallarme-antifurto-domotico-con-home-assistant/). Anzitutto grazie per le ottime guide allo staff di inDomus!! Nella guida (e altrove) si fa riferimento principalmente a sensori di tipo PIR pensati per rilevare movimento in una stanza, oppure a sensori classici a "2 parti" per rilevare l'apertura/chiusura di porte/finestre. Fin qui tutto OK. Tuttavia io sarei interessato a sensori ad effetto "tenda" (es: **LINK RIMOSSO** oppure **LINK RIMOSSO** oppure **LINK RIMOSSO** ). Questi sensori sono utili in quanto triggerano l'allarme PRIMA dell'eventuale irruzione del ladro.

sai che non ho capito?

**Reply by f18m** [reply]
Ciao @KC-inDomus ,
riformulo, mi sono espresso male: volendo realizzare un sistema di allarme usando SOLO HomeAssistant (implementando dentro HA quindi tutta la logica di gestione dei sensori), siete a conoscenza di un buon sensore effetto tenda wireless con anti-tamper?

Qualcosa del tipo:

![image](https://github.com/indomus/forum/assets/9748595/3f0732c9-19a2-4ff5-a103-da9cec5737bd)

Sono sensori che utilizzando PIR oppure PIR+microonde rilevano il movimento in un "piano virtuale" molto sottile, adatto a coprire varchi di accesso come porte e finestre. 
A differenza dei classici sensori magnetici questi consentono di attivare l'allarme prima dell'effrazione (basta che un malitenzionato si avvicini alla porta o finestra, senza nemmeno toccarla e scatta l'allarme!).

In generale non ho trovato nessun sensore per uso domotico (Zigbee, Z-Wave, 433/868 MHz) che abbia le features tipiche di un sensore di allarme "vero":
* anti-tamper
* anti-accecamento/anti-jamming
* bidirectional communication 
* low-battery alarm

Ne conoscete qualcuno?

Grazie ancora

**Reply by indomus** [reply]
valuta quelli [di presenza](https://indomus.it/category/componenti/sensori/presenza/)

**Reply by f18m** [reply]
grazie @indomus per la veloce risposta. Ho guardato tutti i sensori di presenza recensiti e nessuno di essi pare avere le funzionalita' di anti-tamper/anti-accecamento.
Mi domando se sono io che non so cercare o semplicemente e' un settore di mercato in cui i vari vendor di domotica non sono "entrati"... grazie

**Reply by indomus** [reply]
perdonami: un mmWave/Radar, come lo "accechi"?

**Reply by f18m** [reply]
Nel caso dei sensori radar non penso essi siano "accecabili". Tuttavia i sensori domotici di presenza sono (mi pare) tutti per INTERNI. Per esterni un sensore radar/mmWave deve essere di tipo "a tenda" o a "barriera" (con un fascio di microonde molto focalizzato) altrimenti il rischio e' quello di avere continui falsi allarmi.
Il sensore per esterni dovrebbe essere un maggiore deterrente in quanto fa scattare l'allarme PRIMA dell'effrazione. Inoltre il vantaggio di un sensore esterno/a-tenda e' si puo' utilizzare in modalita' "armed_home" ovvero mentre si e' in casa (a patto di non passare dalle porte/finestre ovviamente)... purtroppo in versione "domotica/aperta" non ne ho trovati...

**Comment by marcopi69** [comment]
> Buongiorno, Ho letto la guida inDomus per realizzare un impianto allarme "fai-da-te" tramite HA (https://indomus.it/progetti/realizzare-un-proprio-sistema-dallarme-antifurto-domotico-con-home-assistant/). Anzitutto grazie per le ottime guide allo staff di inDomus!! Nella guida (e altrove) si fa riferimento principalmente a sensori di tipo PIR pensati per rilevare movimento in una stanza, oppure a sensori classici a "2 parti" per rilevare l'apertura/chiusura di porte/finestre. Fin qui tutto OK. Tuttavia io sarei interessato a sensori ad effetto "tenda" (es: **LINK RIMOSSO** oppure **LINK RIMOSSO** oppure **LINK RIMOSSO** ). Questi sensori sono utili in quanto triggerano l'allarme PRIMA dell'eventuale irruzione del ladro.
> 
> Ho cercato un pò ma tuttora non ne ho trovato NESSUNO con standard "aperto" e integrabile in HomeAssistant. Vi chiedo aiuto: siete a conoscenza di un qualche sensore effetto "tenda" o "barriera" installabile in esterno, wireless e integrabile in HomeAssistant?
> 
> Grazie mille in anticipo!

Buongiorno, ti premetto che anche io ho cercato in rete ma non ne sono venuto a capo. Non capisco come nessuno si sia posto il problema. Si trovano solo sensori zigbee o wifi che nella maggior parte dei casi vanno bene per verificare movimento o presenza umana, ma non sanno distinguere una persona da un cane per esempio.
Se non ho capito male vorresti integrare un buon sensore cablato in home assistant, un sensore che distingua una persona da un animale di piccola o media taglia. Sensore di livello professionale che solitamente vengono installati sugli allarmi filari da ditte specializzate che trattano sistemi di allarme.
Una possibile soluzione sarebbe quella di collegare l'uscita cablata del sensore (movimento, presenza e tamper) ad un esp32 che a sua volta trasmette ad home assistant, tramite l'integrazione "esphome", lo stato di ognuno degli ingressi ed attivare le relative automazioni se necessario. A sua volta il modulo esp32 utilizzato è possibile collegarlo ad una interfaccia di rete per avere una connessione stabile.
Non so se mi sono spiegato, comunque allego link ad un articolo dove si può prendere spunto e da cui ho appreso le informazioni che ti ho appena scritto.
edit-no link-

**Reply by f18m** [reply]
ciao @marcopi69 ,
> Se non ho capito male vorresti integrare un buon sensore cablato in home assistant, un sensore che distingua una persona da un animale di piccola o media taglia. Sensore di livello professionale che solitamente vengono installati sugli allarmi filari da ditte specializzate che trattano sistemi di allarme.

Corretto eccetto che mi servirebbero wireless e non cablati. Ma comunque di livello professionale, che generino pochi falsi allarmi e siano pensati per applicazioni di antifurto.

> Una possibile soluzione sarebbe quella di collegare l'uscita cablata del sensore (movimento, presenza e tamper) ad un esp32 che a sua volta trasmette ad home assistant, tramite l'integrazione "esphome", lo stato di ognuno degli ingressi ed attivare le relative automazioni se necessario. A sua volta il modulo esp32 utilizzato è possibile collegarlo ad una interfaccia di rete per avere una connessione stabile.

In effetti penso che questo approccio potrebbe funzionare. Ci sono difficoltà nel farlo ad esempio banalmente bisogna trovare un sensore con abbastanza spazio interno da alloggiare l'ESP32 (ho visto che la schedina classica è di 39 mm x 31.5 mm, non è comunque scontato "farle spazio")... inoltre mi servirebbe trasmettere pure il livello della batteria (ma penso sia fattibile).

Simile a questo approccio, giusto ieri sera ho trovato qualcosa che potrebbe rappresentare una alternativa altamente integrabile in HA senza richiedere lavori di "saldatura" (come l'approccio ESP32 dentro sensore professionale): il "FLYK STA Controller" (evito link) pare essere una interfaccia radio bidirezionale che espone 4 uscite di allarme NC e gestisce fino a 32 sensori professionali che utilizzano il protocollo "FLYK Long Range", come ad esempio il sensore "EXT SLIM". Sia il "FLYK STA Controller" sia il "EXT SLIM" sono della marca "PESS".
Non sono ancora riuscito a trovare però il loro costo (sono in attesa di risposta dal produttore/rivenditore).

**Comment by marcopi69** [comment]
Devi cercare questo argomento: Integrare qualsiasi sistema di allarme in Home Assistant

**Reply by f18m** [reply]
Grazie!

**Comment by marcopi69** [comment]
fammi sapere.....

**Reply by f18m** [reply]
ciao @marcopi69 , ciao tutti

Breve aggiornamento su questo topic: la soluzione "FLYK STA Controller" abbinata ai sensori "EXT SLIM" (tutto marca PESS) penso sarebbe facile da integrare in HA in quanto la scheda "controller" si occupa di dialogare su frequenza radio con protocollo criptato ai vari sensori ed espone dei banali contatti NC all'esterno.
Tuttavia i costi sono elevati (ogni sensore tenda è attorno ai 300€) e ho interrotto la ricerca in tal senso (sensori wireless effetto tenda integrabili in HA).

Difatti l'impianto che voglio realizzare è un impianto di allarme domotico per rimpiazzare un vecchio impianto di cui già dispone una abitazione. Con grande sorpresa, ho scoperto che i sensori tenda "vecchi" già presenti in quella abitazione sono in realtà cablati (semplicemente il cavo era ben nascosto e il sensore copre completamente il corrugato che porta i fili !!!). Questo mi semplifica enormemente l'attività e mi ha reso facile progettare un sistema che utilizza un raspberry in accoppiata ad un HAT con ingressi optoisolati per "leggere" lo stato di tutti i vari contatti NC dei sensori.
Tramite il progetto mqtt-io sono riuscito ad esporre tali letture fatte da Raspberry su topic del broker MQTT che gira sul mio home assistant. Ho poi creato delle entità "binary_sensor" di tipo MQTT. Ora sto provando a seguire la guida https://indomus.it/progetti/realizzare-un-proprio-sistema-dallarme-antifurto-domotico-con-home-assistant/ ...

Spero di aver risolto, grazie a tutti dei suggerimenti intanto!
